pkg/kubelet/status/status_manager.go

```diff
@@ -156,8 +156,7 @@ func (m *manager) Start() {
 	}
 
 	klog.InfoS("Starting to sync pod status with apiserver")
-	//lint:ignore SA1015 Ticker can link since this is only called once and doesn't handle termination.
```

Comment:
does this mean SA1015 is not running any more?

Reply:
I removed the `//lint:ignore` comment because I changed line 160 from `time.Tick`, which leaks (according to the description [here](https://pkg.go.dev/time#Tick)), to `time.NewTicker`, which does not leak. I believe SA1015 is still running though, as it is included in the staticcheck [docs](https://staticcheck.io/docs/checks#SA1015).

Reply:
both would leak if the ticker is not stopped, but i'm assuming that the status manager and this ticker normally have the lifespan of that of the process.

the example here https://pkg.go.dev/time#NewTicker shows how the object should be managed properly.

Reply:
Ah yes you're right. I could add a `defer syncTicker.Stop()` inside that goroutine at line 162 but I'm not sure if that's worth it since the for select loop will run forever anyway, and because as you and the previous //lint:ignore comment both said, the ticker is only created once.

Reply:
process lifespan memory leaks can still show up in software that does leak detection...at least for object code from languages without a GC.

the lack of stop can be a problem, if one day the status_manager code is (for some reason) adapted to a dynamic restart but the stop is never added. maybe our static check tools will catch this problem, maybe not.